### ISSUE TITLE
CRI: Should kubelet or runtime handle Capability ALL?

### ISSUE THREAD

**Issue by Random-Liu** [kubernetes/kubernetes#51980]
`"all"` is not a valid capability string, but docker supports it, e.g. `docker run --cap-drop ALL`.

Since we pass the capabilities strings from K8s API to docker directly, K8s API also supports it. And there are user relying on this now:
https://github.com/microservices-demo/microservices-demo/blob/master/deploy/kubernetes/complete-demo.yaml#L24

The problems are:
1) Should we support this in K8s API? Since we've supported it so far, I doubt we could deprecate it.
2) Should we support this in Kubelet? This is a linux specific thing, but we are doing linux specific thing in Kubelet anyway.
3) Should we support this in container runtime?

Since it's code freeze, I don't think we could update K8s now. We'll support this in cri-containerd for now, and let's revisit this in 1.9.

/cc @ijc Thanks for finding this.
@yujuhong @feiskyer @tallclair @mrunalp 
/cc @kubernetes/sig-node-api-reviews

**Comment by yujuhong** [comment]
> Should we support this in Kubelet? This is a linux specific thing, but we are doing linux specific thing in Kubelet anyway.

+1 for defining the default set of capabilities in the kubernetes codebase. kubelet should translate the kubernetes api to a set of capabilities and pass them to the runtime using CRI.

**Comment by tallclair** [comment]
Yes, we should definitely support dropping all capabilities. I think we should define the api using `'*'` rather than `ALL`, since that's what we use elsewhere in the Kubernetes API (e.g. https://github.com/kubernetes/kubernetes/blob/8823a835655cfcf73ddf8a636e2e11f83baeea57/pkg/apis/extensions/types.go#L889)

**Comment by tallclair** [comment]
To clarify, we should carefully deprecate `ALL` (in favor of `*`) over a few releases to avoid breaking users.
